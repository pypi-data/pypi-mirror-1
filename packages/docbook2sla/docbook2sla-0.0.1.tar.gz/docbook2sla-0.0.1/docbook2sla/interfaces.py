from zope.interface import Interface

class IDocBook2Sla(Interface):

    def convert(docbook_filename, scribus_filename):
        """ Merge a DocBook and a Scribus file (stored on the filesystem) to some output format.
            Returns the filename of the output file.
        """
