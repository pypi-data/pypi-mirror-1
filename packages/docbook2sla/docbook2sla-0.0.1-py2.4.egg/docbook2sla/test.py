from docbook2sla import DocBook2Sla
d2s = DocBook2Sla('xml/content.xml', 'scribus/clean134.sla')
print d2s.convert('xml/content.xml', 'scribus/clean134.sla')

