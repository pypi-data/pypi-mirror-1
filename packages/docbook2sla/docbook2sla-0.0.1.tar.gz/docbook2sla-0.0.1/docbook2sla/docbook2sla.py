import os

from util import newTempfile, runcmd
from logger import LOG

from lxml import etree

from zope.interface import implements
from interfaces import IDocBook2Sla

_debug = 0

class NotValidXmlError(Exception): pass

class DocBook2Sla(object):
    """Conversions from DocBook to Scribus
    """

    implements(IDocBook2Sla)

    def __init__(self, docbookfn, scribusfn):
        self.docbookfn = docbookfn
        self.scribusfn = scribusfn

    def convert(self, docbookfn, scribusfn, output_filename=None):
        """ Converts a docbook and a scribus file
        """
        if not output_filename:
            output_filename = newTempfile(suffix='.sla')

        # create instance of DocBook2Sla
        #d2s = DocBook2Sla(docbookfn, scribusfn)

        # validate input files
        #print 'validate docbook: %s' % d2s.validate(docbookfn)
        #print 'validate scribus: %s' % d2s.validate(scribusfn)

        # add id to every docbook node
        docbookfn_with_ids = self.generateId(docbookfn)
        if _debug:
            print 'docbookfn_with_ids: %s' % docbookfn_with_ids

        # transform docbook into scribus pageobjects
        pageobjectsfn = self.docbook2Pageobjects(docbookfn_with_ids)
        if _debug:
            print 'pageobjectsfn: %s' % pageobjectsfn


        # it is necessary to write the content to a named tmpfile, because
        # the wrapper xslt stylesheet needs a static param input
        self.write(pageobjectsfn, 'content.sla')

        # inject the scribus pageobjects into a scribus wrapper file
        scribusfn = self.wrapper(pageobjectsfn)
        if _debug:
            print 'scribusfn: %s' % scribusfn
        print self.showFile(scribusfn)

        self.write(scribusfn, 'output/output.sla')
        if _debug:
            print 'validate %s: %s' % ('output/output.sla', self.validate(scribusfn))

        print self.analyseScribusFile(scribusfn)

        return output_filename

    def validate(self, fname):
        """ validates a xml against its xml schema
        """

        # split the filename into name and extension
        fe = os.path.splitext( fname )[1]

        # parse the file
        xmltree = etree.parse(fname)

        # docbook or scribus?
        if fe == '.xml':
            # validate docbook
            xmlschema_doc = etree.parse('http://www.docbook.org/xsd/4.5/docbook.xsd')
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        elif fe == '.sla':
            # validate scribus
            xmlschema_doc = etree.parse('schema/scribus.xsd')
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        else:
            raise ValueError('Unsupported format: %s' % fe)

    def generateId(self, docbookfn):
        """ generates ids for every node of a docbook document
        """

        # file -> elementtree
        docbook = etree.parse(docbookfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/generate-id.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(docbook)

        filename = newTempfile(suffix='.xml')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename


    def docbook2Pageobjects(self, docbookfn):
        """ Converts docbook nodes into scribus pageobjects
        """

        # file -> elementtree
        docbook = etree.parse(docbookfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/docbook2pageobject.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(docbook, uselayout="'../scribus/clean134.sla'")

        filename = newTempfile(suffix='.xml')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename


    def wrapper(self, contentfn):
        """ Wraps up Scibus pageobjects into a Scribus document
        """

        foo = "bar"
        # parse the scribus wrapper
        scribusfn = 'scribus/clean134.sla'
        scribuswrapper = etree.parse(scribusfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/wrapper.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(scribuswrapper, content="'../content.sla'")

        filename = newTempfile(suffix='.sla')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename

    def write(self, tmpfn, outputfn):
        """ write tempfile into output file
        """
        tmpFile = open(tmpfn,"r")
        outputFile = open(outputfn,"w")
        outputFile.write(tmpFile.read())
        return outputFile.close()

    def analyseScribusFile(self, scribusfn):
        """ returns statistical informations about a scribus file
        """

        # file -> elementtree
        tree = etree.parse(scribusfn)
        return '# Pageobjects: %s' % len(tree.findall(".//PAGEOBJECT"))

    def analyseDocBookFile(self, docbookfn):
        """ returns statistival informations about a docbook file
        """
        pass

    def showFile(self, fn):
        tmpFile = open(fn,"r")
        return tmpFile.read()

    def __call__(self, *args, **kw):
        return self.convert(*args, **kw)