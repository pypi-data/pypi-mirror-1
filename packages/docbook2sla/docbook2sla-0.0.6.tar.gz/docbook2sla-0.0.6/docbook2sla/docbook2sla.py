#!/usr/bin/python
# -*- coding: utf-8 -*-

__author__ = "Timo Stollenwerk (timo@zmag.de)"
__license__ = "GNU General Public License (GPL)"
__revision__ = "$Rev: 199 $"
__date__ = "$Date: 2008-03-04 10:42:05 +0100 (Di, 04 Mrz 2008) $"
__URL__ = "$URL: https://svn.zmag.de/svn/python/docbook2sla/trunk/docbook2sla/docbook2sla.py $"

import os, sys

from util import newTempfile, runcmd
from logger import LOG

from lxml import etree

from zope.interface import implements
from interfaces import IDocBook2Sla

_debug = 1

class NotValidXmlError(Exception): pass

class DocBook2Sla(object):
    """Conversions from DocBook to Scribus. """

    implements(IDocBook2Sla)

    def create(self, docbookfn, scribusfn, output_filename=None, validate=False):
        """ Creates a Scribus file from a DocBook source. """

        if _debug:
            print"---- create ----"

        # validate input files
        if validate:
            print 'validate docbook: %s' % self.validate(docbookfn)
            print 'validate scribus: %s' % self.validate(scribusfn)

        # generate id : content.xml -> content.id.xml
        docbook_with_ids = self.transform(docbookfn, 'xsl/generate-id.xsl')

        #return open(docbook_with_ids, 'r').read()

        # docbook2pageobject : 01_content.id.xml -> pageobjects.xml
        scribus_pageobjects = self.transform(docbook_with_ids,
                                             'xsl/docbook2pageobject.xsl')

        #return open(scribus_pageobjects, 'r').read()

        # wrapper : pageobjects.xml + clean134.sla -> output.sla
        output = self.transform(scribusfn,
                                'xsl/wrapper.xsl',
                                secondinputfn=scribus_pageobjects,
                                )

        #return open(output, 'r').read()

        # write to output filename if specified
        if output_filename:
            open(output_filename, 'w').write(open(output, 'r').read())

        return output_filename

    def syncronize(self, docbookfn, scribusfn, output_filename=None, validate=False):
        """ Syncronize changes in a docbook file into a scribus file. """

        if _debug:
            print"---- syncronize ----"

        # validate input files
        if validate:
            try:
                print 'validate docbook: %s' % self.validate(docbookfn)
                print 'validate scribus: %s' % self.validate(scribusfn)
            except:
                print >>sys.stderr, 'Error validating %s and %s' % (docbookfn, scribusfn)
                raise

        # generate id : content.xml -> content.id.xml
        docbook_with_ids = self.transform(docbookfn, 'xsl/generate-id.xsl')

        # docbook2pageobject : 01_content.id.xml -> pageobjects.xml
        scribus_pageobjects = self.transform(docbook_with_ids,
                                             'xsl/docbook2pageobject.xsl')

        # wrapper : pageobjects.xml + clean134.sla -> output.sla
        output = self.transform(scribusfn,
                                'xsl/wrapper.xsl',
                                secondinputfn=scribus_pageobjects,
                                )

        # write to output filename if specified
        if output_filename:
            open(output_filename, 'w').write(open(output, 'r').read())

        return output_filename

    def transform(self, xmlfn, stylefn, secondinputfn=None, outputfn=None, validate=False):
        """ Transforms a XML document with a XSL stylesheet. """

        # if no output filename is specified, create tempfile
        if not outputfn:
            outputfn = newTempfile(suffix='.xml')

        # try to transform inputfn
        try:
            # xml -> elementtree
            xmltree = etree.parse(xmlfn)

            # parse the xslt stylesheet
            styletree = etree.parse(stylefn)

            # load the xslt stylesheet
            xslt = etree.XSLT(styletree)

            if secondinputfn:

                # get the stylesheet path
                pathname = os.path.dirname(stylefn)
                abspath = os.path.abspath(pathname)
                tmpfile = "%s/tmp.xml" % abspath

                # it is necessary to write the content to a named tmpfile, because
                # the wrapper xslt stylesheet needs a static param input
                # ToDo: named tmpfile
                self.write(secondinputfn, tmpfile)

                # transform the xml file with the xslt stylesheet
                result = xslt(xmltree, secondinput="'tmp.xml'")

                # delete content.sla file
                os.remove(tmpfile)
            else:
                result = xslt(xmltree)

            if result != None:
                # write output file
                open(outputfn, 'w').write(etree.tostring(result.getroot()))

        except:
            print >>sys.stderr, 'Error transforming %s (%s) with %s' % (xmlfn, secondinputfn, stylefn)
            raise

        # return the output filename
        return outputfn

    def validate(self, fname):
        """ Validates XML against DocBook or Scribus XML Schema. """

        # split the filename into name and extension
        fe = os.path.splitext( fname )[1]

        # parse the file
        xmltree = etree.parse(fname)

        # docbook or scribus?
        if fe == '.xml':
            # validate docbook
            xmlschema_doc = etree.parse('http://www.docbook.org/xsd/4.5/docbook.xsd')
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        elif fe == '.sla':
            # validate scribus
            xmlschema_doc = etree.parse('schema/scribus.xsd')
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        else:
            raise ValueError('Unsupported format: %s' % fe)

    def generateId(self, docbookfn):
        """ Generates IDs for every node of a DocBook file. """

        # file -> elementtree
        docbook = etree.parse(docbookfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/generate-id.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(docbook)

        filename = newTempfile(suffix='.xml')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename


    def docbook2Pageobjects(self, docbookfn):
        """ Converts DocBook nodes into Scribus pageobjects. """

        # file -> elementtree
        docbook = etree.parse(docbookfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/docbook2pageobject.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(docbook, uselayout="'tests/data/scribus/clean134.sla'")

        filename = newTempfile(suffix='.xml')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename


    def wrapper(self, contentfn, scribusfn):
        """ Wraps up Scibus pageobjects with a Scribus document. """

        # parse the scribus wrapper
        scribuswrapper = etree.parse(scribusfn)

        # parse the xslt stylesheet
        style = etree.parse('xsl/wrapper.xsl')

        # load the xslt stylesheet
        xslt = etree.XSLT(style)

        # transform the xml file with the xslt stylesheet
        result = xslt(scribuswrapper, content="'../content.sla'")

        filename = newTempfile(suffix='.sla')
        open(filename, 'wb').write(etree.tostring(result.getroot()))
        return filename

    def write(self, tmpfn, outputfn):
        """ Write tempfile into output file. """
        tmpFile = open(tmpfn,"r")
        outputFile = open(outputfn,"w")
        outputFile.write(tmpFile.read())
        return outputFile.close()

    def analyseScribusFile(self, scribusfn):
        """ Returns statistical informations about a Scribus file. """

        # file -> elementtree
        tree = etree.parse(scribusfn)
        return '# Pageobjects: %s' % len(tree.findall(".//PAGEOBJECT"))

    def analyseDocBookFile(self, docbookfn):
        """ Returns statistival informations about a DocBook file. """
        pass

    def showFile(self, fn):
        """ Returns string representation of a file. """
        tmpFile = open(fn,"r")
        return tmpFile.read()

    def __call__(self, *args, **kw):
        return self.create(*args, **kw)