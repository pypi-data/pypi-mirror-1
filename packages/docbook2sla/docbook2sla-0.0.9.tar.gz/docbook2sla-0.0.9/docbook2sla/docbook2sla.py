#!/usr/bin/python
# -*- coding: utf-8 -*-

__author__ = "Timo Stollenwerk (timo@zmag.de)"
__license__ = "GNU General Public License (GPL)"
__revision__ = "$Rev: 225 $"
__date__ = "$Date: 2008-03-07 09:23:44 +0100 (Fri, 07 Mar 2008) $"
__URL__ = "$URL: http://svn.zmag.de/svn/python/docbook2sla/trunk/docbook2sla/docbook2sla.py $"

import os, sys

from util import newTempfile, runcmd
from logger import LOG

from lxml import etree

from zope.interface import implements
from interfaces import IDocBook2Sla

dirname = os.path.dirname(__file__)

_debug = 1

class NotValidXmlError(Exception): pass

class DocBook2Sla(object):
    """Conversions from DocBook to Scribus. """

    implements(IDocBook2Sla)

    def __init__(self):

        # make stylesheets available
        xsl_generateid = os.path.abspath(os.path.join(dirname, 'xsl', 'generate-id.xsl'))
        if not os.path.exists(xsl_generateid):
            raise IOError('%s does not exist' % xsl_generateid)

        xsl_docbook2pageobject = os.path.abspath(os.path.join(dirname, 'xsl', 'docbook2pageobject.xsl'))
        if not os.path.exists(xsl_docbook2pageobject):
            raise IOError('%s does not exist' % xsl_docbook2pageobject)

        xsl_wrapper = os.path.abspath(os.path.join(dirname, 'xsl', 'wrapper.xsl'))
        if not os.path.exists(xsl_wrapper):
            raise IOError('%s does not exist' % xsl_wrapper)

        self.xsl_generateid = xsl_generateid
        self.xsl_docbook2pageobject = xsl_docbook2pageobject
        self.xsl_wrapper = xsl_wrapper

    def create(self, docbookfn, scribusfn, output_filename=None, validate=False):
        """ Creates a Scribus file from a DocBook source. """

        if _debug:
            print"---- create ----"

        # validate input files
        if validate:
            print 'validate docbook: %s' % self.validate(docbookfn)
            print 'validate scribus: %s' % self.validate(scribusfn)


        # generate id : content.xml -> content.id.xml
        docbook_with_ids = self.transform(docbookfn, self.xsl_generateid)

        if _debug:
            print "DEBUG: docbook_with_ids written to '%s'" % docbook_with_ids
            #return open(docbook_with_ids, 'r').read()

        # docbook2pageobject : 01_content.id.xml -> pageobjects.xml
        scribus_pageobjects = self.transform(docbook_with_ids, self.xsl_docbook2pageobject)

        if _debug:
            print "DEBUG: scribus_pageobjects written to '%s'" % scribus_pageobjects
            #return open(scribus_pageobjects, 'r').read()

        # wrapper : pageobjects.xml + clean134.sla -> output.sla
        output = self.transform(scribusfn, self.xsl_wrapper, secondinputfn=scribus_pageobjects)

        if _debug:
            print "DEBUG: output written to '%s'" % output
            #return open(output, 'r').read()

        # write to output filename if specified
        if output_filename:
            open(output_filename, 'w').write(open(output, 'r').read())
            return output_filename
        else:
            return output

    def syncronize(self, docbookfn, scribusfn, output_filename=None, validate=False):
        """ Syncronize changes in a docbook file into a scribus file. """

        if _debug:
            print"---- syncronize ----"

        # validate input files
        if validate:
            try:
                print 'validate docbook: %s' % self.validate(docbookfn)
                print 'validate scribus: %s' % self.validate(scribusfn)
            except:
                print >>sys.stderr, 'Error validating %s and %s' % (docbookfn, scribusfn)
                raise

        # generate id : content.xml -> content.id.xml
        docbook_with_ids = self.transform(docbookfn, self.xsl_generateid)

        if _debug:
            print "DEBUG: docbook_with_ids written to '%s'" % docbook_with_ids
            #return open(docbook_with_ids, 'r').read()

        # docbook2pageobject : 01_content.id.xml -> pageobjects.xml
        scribus_pageobjects = self.transform(docbook_with_ids, self.xsl_docbook2pageobject)

        if _debug:
            print "DEBUG: scribus_pageobjects written to '%s'" % scribus_pageobjects
            #return open(scribus_pageobjects, 'r').read()

        # wrapper : pageobjects.xml + clean134.sla -> output.sla
        output = self.transform(scribusfn, xsl_wrapper, secondinputfn=scribus_pageobjects)

        if _debug:
            print "DEBUG: output written to '%s'" % output
            #return open(output, 'r').read()

        # write to output filename if specified
        if output_filename:
            open(output_filename, 'w').write(open(output, 'r').read())
            return output_filename
        else:
            return output

    def transform(self, xmlfn, stylefn, secondinputfn=None, outputfn=None, validate=False):
        """ Transforms a XML document with a XSL stylesheet. """

        # if no output filename is specified, create tempfile
        if not outputfn:
            outputfn = newTempfile(suffix='.xml')

        # try to transform inputfn
        try:
            # xml -> elementtree
            xmltree = etree.parse(xmlfn)

            # parse the xslt stylesheet
            styletree = etree.parse(stylefn)

            # load the xslt stylesheet
            xslt = etree.XSLT(styletree)

            if secondinputfn:

                # get the stylesheet path
                pathname = os.path.dirname(stylefn)
                abspath = os.path.abspath(pathname)
                tmpfile = "%s/tmp.xml" % abspath

                # it is necessary to write the content to a named tmpfile, because
                # the wrapper xslt stylesheet needs a static param input
                # ToDo: named tmpfile
                self.write(secondinputfn, tmpfile)

                # transform the xml file with the xslt stylesheet
                result = xslt(xmltree, secondinput="'tmp.xml'")

                # delete content.sla file
                os.remove(tmpfile)
            else:
                result = xslt(xmltree)

            if result != None:
                # write output file
                open(outputfn, 'w').write(etree.tostring(result.getroot()))

        except:
            print >>sys.stderr, 'Error transforming %s (%s) with %s' % (xmlfn, secondinputfn, stylefn)
            raise

        # return the output filename
        return outputfn

    def validate(self, fname):
        """ Validates XML against DocBook or Scribus XML Schema. """


        # split the filename into name and extension
        fe = os.path.splitext( fname )[1]

        # parse the file
        xmltree = etree.parse(fname)

        # docbook or scribus?
        if fe == '.xml':
            # validate docbook
            xmlschema_doc = etree.parse('http://www.docbook.org/xsd/4.5/docbook.xsd')
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        elif fe == '.sla':
            # validate scribus

            # make scribus file available
            xml_schema_scribus = os.path.abspath(os.path.join(dirname, 'schema', 'scribus.xsd'))
            if not os.path.exists(xml_schema_scribus):
                raise IOError('%s does not exist' % xml_schema_scribus)

            xmlschema_doc = etree.parse(xml_schema_scribus)
            xmlschema = etree.XMLSchema(xmlschema_doc)
            if xmlschema.validate(xmltree):
                return True
            else:
                raise NotValidXmlError, "%s is no valid %s file." % (fname, fe)
        else:
            raise ValueError('Unsupported format: %s' % fe)

    def write(self, tmpfn, outputfn):
        """ Write tempfile into output file. """
        tmpFile = open(tmpfn,"r")
        outputFile = open(outputfn,"w")
        outputFile.write(tmpFile.read())
        return outputFile.close()

    def analyseScribusFile(self, scribusfn):
        """ Returns statistical informations about a Scribus file. """

        # file -> elementtree
        tree = etree.parse(scribusfn)
        return '# Pageobjects: %s' % len(tree.findall(".//PAGEOBJECT"))

    def analyseDocBookFile(self, docbookfn):
        """ Returns statistival informations about a DocBook file. """
        pass

    def showFile(self, fn):
        """ Returns string representation of a file. """
        tmpFile = open(fn,"r")
        return tmpFile.read()

    def __call__(self, *args, **kw):
        return self.create(*args, **kw)