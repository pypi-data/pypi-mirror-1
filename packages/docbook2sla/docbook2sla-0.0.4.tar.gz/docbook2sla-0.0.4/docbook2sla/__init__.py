from interfaces import IDocBook2Sla
from docbook2sla import DocBook2Sla