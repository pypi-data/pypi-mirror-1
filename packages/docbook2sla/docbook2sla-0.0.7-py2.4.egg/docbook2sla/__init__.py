#!/usr/bin/python
# -*- coding: utf-8 -*-

__author__ = "Timo Stollenwerk (timo@zmag.de)"
__license__ = "GNU General Public License (GPL)"
__revision__ = "$Rev: 178 $"
__date__ = "$Date: 2008-03-01 23:21:48 +0100 (Sa, 01 Mrz 2008) $"
__URL__ = "$URL: https://svn.zmag.de/svn/python/docbook2sla/trunk/docbook2sla/__init__.py $"

from interfaces import IDocBook2Sla
from docbook2sla import DocBook2Sla