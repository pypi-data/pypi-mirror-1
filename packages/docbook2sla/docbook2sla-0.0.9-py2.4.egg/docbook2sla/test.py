from docbook2sla import DocBook2Sla
d2s = DocBook2Sla()


print "=== test with named output files ==="
print d2s.create('tests/data/xml/article.xml',\
                 'tests/data/scribus/clean134.sla',\
                 output_filename='tests/data/output/create_output.sla')
print d2s.syncronize('tests/data/xml/content-1.xml',\
                     'tests/data/output/create_output.sla',\
                     output_filename='tests/data/output/syncronize_output.sla')

print "=== test with unnamed output files ==="
print d2s.create('tests/data/xml/article.xml',\
                 'tests/data/scribus/clean134.sla')
print d2s.syncronize('tests/data/xml/content-1.xml',\
                     'tests/data/output/create_output.sla')


print "=== test with named output files ==="
print d2s.create('tests/data/xml/content.xml',\
                 'tests/data/scribus/clean134.sla',\
                 output_filename='tests/data/output/create_output.sla')
print d2s.syncronize('tests/data/xml/content-1.xml',\
                     'tests/data/output/create_output.sla',\
                     output_filename='tests/data/output/syncronize_output.sla')

print "=== test with unnamed output files ==="
print d2s.create('tests/data/xml/content.xml',\
                 'tests/data/scribus/clean134.sla')
print d2s.syncronize('tests/data/xml/content-1.xml',\
                     'tests/data/output/create_output.sla')
