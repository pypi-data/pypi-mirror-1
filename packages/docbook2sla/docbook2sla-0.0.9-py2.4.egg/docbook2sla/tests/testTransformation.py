import os
import sys
import unittest

import tempfile

from lxml import etree
from StringIO import StringIO

from docbook2sla import DocBook2Sla

dirname = os.path.dirname(__file__)

class TransformationTestCase(unittest.TestCase):

    def setUp(self):
        pass


    def testTransformation(self):

        inputfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformation.input.xml')
        transformfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformation.xsl')
        expectedfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformation.expected.xml')

        print >>sys.stderr, 'testSimpleTransformation: %s, %s' % (inputfn, transformfn)

        d2s = DocBook2Sla()

        # try to transform inputfn
        try:
            outputfn = d2s.transform(inputfn, transformfn)
        except:
            print >>sys.stderr, 'Error transforming %s with %s' % (inputfn, transformfn)
            raise

        # open output
        output = open(outputfn, 'r')

        # open expected output
        expected = open(expectedfn, 'r')

        self.assertEqual(output.read(), expected.read())

        return outputfn


    def testTransformationWithParam(self):
        """ Test transformation with parameters passed.
            xsltproc \
            --stringparam secondinput testTransformationWithParam.secondinput.xml \
            tests/data/testTransformationWithParam.xsl \
            tests/data/testTransformationWithParam.input.xml
        """

        inputfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformationWithParam.input.xml')
        secondinputfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformationWithParam.secondinput.xml')
        transformfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformationWithParam.xsl')
        expectedfn = os.path.join(os.path.dirname(__file__), 'data', 'testTransformationWithParam.expected.xml')

        print >>sys.stderr, 'testTransformationWithParam: %s, %s' % (inputfn, transformfn)

        d2s = DocBook2Sla()

        # try to transform inputfn
        try:
            outputfn = d2s.transform(inputfn, transformfn, secondinputfn=secondinputfn)

        except:
            print >>sys.stderr, 'Error transforming %s with %s' % (inputfn, transformfn)
            raise

        # open output
        output = open(outputfn, 'r').read()

        # open expected output
        expected = open(expectedfn, 'r').read()

        self.assertEqual(output, expected)

        return outputfn


    def testDocBook2PageobjectsArticle(self):
        """ Test transformation from DocBook article to Scribus pageobjects. """

        xsl_docbook2pageobject = os.path.abspath(os.path.join(dirname,
                                                              '..',
                                                              'xsl',
                                                              'docbook2pageobject.xsl'))
        if not os.path.exists(xsl_docbook2pageobject):
            raise IOError('%s does not exist' % xsl_docbook2pageobject)

        d2s = DocBook2Sla()
        outputfn = d2s.transform('tests/data/xml/article.xml',
                                 xsl_docbook2pageobject)
        output = open(outputfn, 'r').read()
        outputtree = etree.XML(output)
        count_pageobjects = outputtree.xpath("count(//PAGEOBJECT)")
        self.assertEqual(count_pageobjects, 4)


    def testDocBook2PageobjectsBook(self):
        """ Test transformation from DocBook book to Scribus pageobjects. """

        xsl_docbook2pageobject = os.path.abspath(os.path.join(dirname,
                                                              '..',
                                                              'xsl',
                                                              'docbook2pageobject.xsl'))
        if not os.path.exists(xsl_docbook2pageobject):
            raise IOError('%s does not exist' % xsl_docbook2pageobject)

        d2s = DocBook2Sla()
        outputfn = d2s.transform('tests/data/xml/book.xml',
                                 xsl_docbook2pageobject)
        output = open(outputfn, 'r').read()
        outputtree = etree.XML(output)
        count_pageobjects = outputtree.xpath("count(//PAGEOBJECT)")
        self.assertEqual(count_pageobjects, 4.0)


    def testWrapper(self):
        """ Test the wrapper function. """

        d2s = DocBook2Sla()

        inputfn = os.path.join(os.path.dirname(__file__), 'data', 'testWrapper.input.xml')
        secondinputfn = os.path.join(os.path.dirname(__file__), 'data', 'testWrapper.secondinput.xml')
        transformfn = os.path.abspath(os.path.join(dirname, '..', 'xsl', 'wrapper.xsl'))

        outputfn = d2s.transform(inputfn, transformfn, secondinputfn=secondinputfn)
        output = open(outputfn, 'r').read()
        outputtree = etree.XML(output)

        # make sure the first child of the root node is SCRIBUSUTF8NEW
        root_node = outputtree.xpath("name(/*)")
        self.assertEqual(root_node, "SCRIBUSUTF8NEW")

        # make sure the number of pageobjects is correct
        count_pageobjects = outputtree.xpath("count(//PAGEOBJECT)")
        self.assertEqual(count_pageobjects, 2.0)

        # make sure there is a pageobject with id1
        po1 = outputtree.xpath("name(//PAGEOBJECT[@ANNAME='id1'])")
        self.assertEqual(po1, "PAGEOBJECT")

        # make sure there is a pageobject with id2
        po1 = outputtree.xpath("name(//PAGEOBJECT[@ANNAME='id2'])")
        self.assertEqual(po1, "PAGEOBJECT")



    def testCreateArticle(self):
        """ Test create() function with a DocBook article """

        d2s = DocBook2Sla()
        outputfn = d2s.create('tests/data/xml/article.xml',
                              'tests/data/scribus/clean134.sla')
        output = open(outputfn, 'r').read()
        outputtree = etree.XML(output)

        # make sure the first child of the root node is SCRIBUSUTF8NEW
        root_node = outputtree.xpath("name(/*)")
        self.assertEqual(root_node, "SCRIBUSUTF8NEW")

        count_pageobjects = outputtree.xpath("count(//PAGEOBJECT)")
        self.assertEqual(count_pageobjects, 4.0)


    def testCreateBook(self):
        """ Test create() function with a DocBook book """
        d2s = DocBook2Sla()
        outputfn = d2s.create('tests/data/xml/book.xml',
                              'tests/data/scribus/clean134.sla')
        output = open(outputfn, 'r').read()
        outputtree = etree.XML(output)
        count_pageobjects = outputtree.xpath("count(//PAGEOBJECT)")
        self.assertEqual(count_pageobjects, 4.0)

#    def testSyncronize(self):
#        d2s = DocBook2Sla()
#        print d2s.syncronize('tests/data/xml/article-1.xml',
#                             'tests/data/output/create_output.sla',
#                             'tests/data/output/syncronize_output.sla')

def test_suite():
    suite = unittest.TestSuite()
    suite.addTest(makeSuite(TransformationTestCase))
    return suite

if __name__ == "__main__":
    unittest.main()