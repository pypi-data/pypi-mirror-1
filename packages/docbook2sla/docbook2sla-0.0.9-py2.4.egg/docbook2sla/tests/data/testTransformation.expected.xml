<bar><foo>test</foo></bar>